src/net/SDP/SDP.cs

```diff
@@ -664,6 +667,39 @@ public static SDP ParseSDPDescription(string sdpDescription)
                                 }
                                 break;
 
+                            case var l when l.StartsWith(SDPMediaAnnouncement.MEDIA_FORMAT_PATH_ACCEPT_TYPES_PREFIX):
+                                if (activeAnnouncement != null)
+                                {
+                                    string acceptTypesStr = sdpLineTrimmed.Substring(sdpLineTrimmed.IndexOf(':') + 1);
+                                    var acceptTypesList = acceptTypesStr.Trim().Split(' ').ToList();
+                                    activeAnnouncement.MessageMediaFormat.AcceptTypes = acceptTypesList;
+                                }
+                                else
+                                {
+                                    logger.LogWarning("A accept-types attribute can only be set on a media announcement.");
+                                }
+                                break;
+
+                            case var l when l.StartsWith(SDPMediaAnnouncement.MEDIA_FORMAT_PATH_MSRP_PREFIX):
+                                if (activeAnnouncement != null)
+                                {
+                                    string pathStr = sdpLineTrimmed.Substring(sdpLineTrimmed.IndexOf(':') + 1);
+                                    string pathTrimmedStr = pathStr.Substring(pathStr.IndexOf(':') + 3);
+                                    activeAnnouncement.MessageMediaFormat.IP = pathTrimmedStr.Substring(0, pathTrimmedStr.IndexOf(':'));
+                                    
+                                    pathTrimmedStr = pathTrimmedStr.Substring(pathTrimmedStr.IndexOf(':') + 1);
+                                    activeAnnouncement.MessageMediaFormat.Port = pathTrimmedStr.Substring(0, pathTrimmedStr.IndexOf('/'));
+                                    
+                                    pathTrimmedStr = pathTrimmedStr.Substring(pathTrimmedStr.IndexOf('/') + 1);
+                                    activeAnnouncement.MessageMediaFormat.Endpoint = pathTrimmedStr;

```

Comment:
``` csharp
var uri = new Uri("path:msrp://alice.example.com:7394/2s93i9ek2a;tcp");

Console.WriteLine($"AbsolutePath: {uri.AbsolutePath}");
Console.WriteLine($"LocalPath: {uri.LocalPath}");
Console.WriteLine($"OriginalString: {uri.OriginalString}");
Console.WriteLine($"PathAndQuery: {uri.PathAndQuery}");
Console.WriteLine($"Port: {uri.Port}");
Console.WriteLine($"Scheme: {uri.Scheme}");
Console.WriteLine($"Segments: {string.Join(", ", uri.Segments)}");
```

---
AbsolutePath: /2s93i9ek2a;tcp
LocalPath: /2s93i9ek2a;tcp
OriginalString: msrp://alice.example.com:7394/2s93i9ek2a;tcp
PathAndQuery: /2s93i9ek2a;tcp
Port: 7394
Scheme: msrp
Segments: /, 2s93i9ek2a;tcp
---

Would Uri be a better choice?

Reply:
Or stripping the path: from the start then using Uri

``` csharp
var uri = new Uri("msrp://alice.example.com:7394/2s93i9ek2a;tcp");

Console.WriteLine($"AbsolutePath: {uri.AbsolutePath}");
Console.WriteLine($"AbsoluteUri: {uri.AbsoluteUri}");
Console.WriteLine($"DnsSafeHost: {uri.DnsSafeHost}");
Console.WriteLine($"Fragment: {uri.Fragment}");
Console.WriteLine($"Host: {uri.Host}");
Console.WriteLine($"HostNameType: {uri.HostNameType}");
Console.WriteLine($"IdnHost: {uri.IdnHost}");
Console.WriteLine($"IsAbsoluteUri: {uri.IsAbsoluteUri}");
Console.WriteLine($"IsDefaultPort: {uri.IsDefaultPort}");
Console.WriteLine($"IsFile: {uri.IsFile}");
Console.WriteLine($"IsLoopback: {uri.IsLoopback}");
Console.WriteLine($"IsUnc: {uri.IsUnc}");
Console.WriteLine($"LocalPath: {uri.LocalPath}");
Console.WriteLine($"OriginalString: {uri.OriginalString}");
Console.WriteLine($"PathAndQuery: {uri.PathAndQuery}");
Console.WriteLine($"Port: {uri.Port}");
Console.WriteLine($"Query: {uri.Query}");
Console.WriteLine($"Scheme: {uri.Scheme}");
Console.WriteLine($"Segments: {string.Join(", ", uri.Segments)}");
```

AbsolutePath: /2s93i9ek2a;tcp
AbsoluteUri: msrp://alice.example.com:7394/2s93i9ek2a;tcp
DnsSafeHost: alice.example.com
Fragment: 
Host: alice.example.com
HostNameType: Dns
IdnHost: alice.example.com
IsAbsoluteUri: True
IsDefaultPort: False
IsFile: False
IsLoopback: False
IsUnc: False
LocalPath: /2s93i9ek2a;tcp
OriginalString: msrp://alice.example.com:7394/2s93i9ek2a;tcp
PathAndQuery: /2s93i9ek2a;tcp
Port: 7394
Query: 
Scheme: msrp
Segments: /, 2s93i9ek2a;tcp

Reply:
@Yitzchok changes follow the convention in this file. for sure there are many implementations we can refactor it in next round